collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderList.java

```diff
@@ -156,29 +162,39 @@ public void onUploadButtonsClicked(Button button) {
                     Context.CONNECTIVITY_SERVICE);
             NetworkInfo ni = connectivityManager.getActiveNetworkInfo();
 
-            if (NetworkReceiver.running) {
-                ToastUtils.showShortToast(R.string.send_in_progress);
-                return;
-            } else if (ni == null || !ni.isConnected()) {
+            if (ni == null || !ni.isConnected()) {
                 logger.logAction(this, "uploadButton", "noConnection");
                 ToastUtils.showShortToast(R.string.no_connection);
                 return;
             }
-        }
 
-        int checkedItemCount = getCheckedCount();
-        logger.logAction(this, "uploadButton", Integer.toString(checkedItemCount));
+            LiveData<List<WorkStatus>> statuses = WorkManager.getInstance().getStatusesForUniqueWork(AutoSendWorker.TAG);
 
-        if (checkedItemCount > 0) {
-            // items selected
-            uploadSelectedFiles(button.getId());
-            setAllToCheckedState(listView, false);
-            toggleButtonLabel(findViewById(R.id.toggle_button), listView);
-            uploadButton.setEnabled(false);
-            smsUploadButton.setEnabled(false);
-        } else {
-            // no items selected
-            ToastUtils.showLongToast(R.string.noselect_error);
+            statuses.observe(this, workStatuses -> {
+                if (workStatuses != null) {
+                    for (WorkStatus status : workStatuses) {
+                        if (status.getState().equals(State.RUNNING)) {
+                            ToastUtils.showShortToast(R.string.send_in_progress);
+                            return;
+                        }
+                    }
+                }
+
+                int checkedItemCount = getCheckedCount();
```

Comment:
Why is this in the statuses observer?

Reply:
What did you try to make it synchronous?

I believe this does what you want:
```
            List<WorkStatus> statuses = WorkManager.getInstance().getStatusesForUniqueWork(AutoSendWorker.TAG).getValue();

            if (statuses != null) {
                for (WorkStatus status : statuses) {
                    if (status.getState().equals(State.RUNNING)) {
                        ToastUtils.showShortToast(R.string.send_in_progress);
                        return;
                    }
                }
            }

            int checkedItemCount = getCheckedCount();
            logger.logAction(this, "uploadButton", Integer.toString(checkedItemCount));

            if (checkedItemCount > 0) {
                // items selected
                uploadSelectedFiles(button.getId());
                setAllToCheckedState(listView, false);
                toggleButtonLabel(findViewById(R.id.toggle_button), listView);
                uploadButton.setEnabled(false);
                smsUploadButton.setEnabled(false);
            } else {
                // no items selected
                ToastUtils.showLongToast(R.string.noselect_error);
            }
```

If you wanted to actually use `LiveData`, the structure would need to change quite a bit.

Reply:
To provide an update here -- the `LiveData` object may not immediately have a value to get so the approach I proposed won't work. Really, the issue with the original code is that the observer runs for the entire life time of `InstanceUploaderList`. @jd-alexander is trying an approach that registers an observer in `onCreate` and uses it to set a flag that keeps track of whether or not an auto update is ongoing. That flag can then be read in the click handler.

If anyone has more experience with these frameworks and has an alternate suggestion, please jump in!